services/api/src/api/routes/croissant.py

```diff
@@ -0,0 +1,238 @@
+import logging
+from collections.abc import Mapping
+from http import HTTPStatus
+from itertools import islice
+from typing import Any, Optional, TypedDict
+
+from datasets import Features, Image, Value
+from libapi.authentication import auth_check
+from libapi.exceptions import (
+    ApiError,
+    MissingRequiredParameterError,
+    UnexpectedApiError,
+)
+from libapi.request import get_request_parameter
+from libapi.utils import (
+    Endpoint,
+    are_valid_parameters,
+    get_cache_entry_from_steps,
+    get_json_api_error_response,
+    get_json_error_response,
+    get_json_ok_response,
+)
+from libcommon.processing_graph import ProcessingGraph
+from libcommon.prometheus import StepProfiler
+from starlette.requests import Request
+from starlette.responses import Response
+
+MAX_CONFIGS = 100
+
+
+class ParquetFile(TypedDict):
+    dataset: str
+    config: str
+    split: str
+    url: str
+    filename: str
+    size: int
+
+
+HF_TO_CRROISSANT_VALUE_TYPE = {
+    "string": "sc:Text",
+    "int32": "sc:Integer",
+    "int64": "sc:Integer",
+    "float32": "sc:Float",
+    "float64": "sc:Float",
+    "bool": "sc:Boolean",
+}
+
+
+def get_croissant_from_dataset_infos(dataset: str, infos: list[Mapping[str, Any]], partial: bool) -> Mapping[str, Any]:
+    distribution = [
+        {
+            "@type": "sc:FileObject",
+            "name": "repo",
+            "description": "The Hugging Face git repository.",
+            "contentUrl": f"https://huggingface.co/datasets/{dataset}",
+            "encodingFormat": "git+https",
+            "sha256": "https://github.com/mlcommons/croissant/issues/80",
```

Comment:
Is this a TODO?

Reply:
Yes it will be replaced by actual hashes later, once it's supported in mlcroissant